src/storage/storage.android.ts

```diff
@@ -85,7 +85,7 @@ export function uploadFile(arg: UploadFileOptions): Promise<UploadFileResult> {
 
       const onFailureListener = new gmsTasks.OnFailureListener({
         onFailure: exception => {
-          reject("Upload failed. " + exception);
+          reject(exception);
```

Comment:
I'm not sure it's wise to return the native Java Exception object here. It used to be a String representation. Not saying that's perfect, but this is not better imo.

Reply:
I'm not sure either, that's why it's a rough fix. But if not the entire object, the Google error code may still work.